core/src/test/java/com/linecorp/armeria/common/ResponsePeekTest.java

```diff
@@ -48,25 +49,127 @@ void peekHeaders() {
     }
 
     @Test
-    void peekHeadersWithMapChain() {
+    void peekHeadersThrow() {
+        final HttpResponse res = HttpResponse.of(ResponseHeaders.of(HttpStatus.OK), HttpData.ofUtf8("foo"));
+
+        assertThatThrownBy(() -> res
+                .peekHeaders(headers -> {
+                    if (HttpStatus.OK.equals(headers.status())) {
+                        throw new IllegalStateException();
+                    }
+                })
+                .peekHeaders(headers -> peekCount.incrementAndGet())
+                .aggregate().join())
+                .isInstanceOf(CompletionException.class)
+                .hasCauseInstanceOf(IllegalStateException.class);
+        assertThat(peekCount).hasValue(0);
+    }
+
+    @Test
+    void peekData() {
+        final ResponseHeaders headers = ResponseHeaders.of(HttpStatus.OK);
+        final HttpResponse res = HttpResponse.of(headers,
+                                                 HttpData.ofUtf8("foo"),
+                                                 HttpData.ofUtf8("bar"));
+        final HttpResponse transformed = res
+                .peekData(data -> assertThat(data.toStringUtf8()).doesNotContain("\n"))
+                .mapData(data -> HttpData.ofUtf8(data.toStringUtf8() + '\n'))
+                .peekData(data -> assertThat(data.toStringUtf8()).contains("\n"));
+
+        final AggregatedHttpResponse aggregated = transformed.aggregate().join();
+        assertThat(aggregated.contentUtf8()).isEqualTo("foo\nbar\n");
+        assertThat(aggregated.headers().toBuilder().removeAndThen(HttpHeaderNames.CONTENT_LENGTH).build())
```

Comment:
Because we didn't modify the `ResponseHeaders` in the test, so I think we don't need this assertion.